packages/aws-cdk-lib/aws-s3/lib/bucket.ts

```diff
@@ -2460,6 +2609,122 @@ export class Bucket extends BucketBase {
+    }
+    if (props.replicationRules.length > 1 && props.replicationRules.some(rule => rule.priority === undefined)) {
+      throw new Error('\'priority\' must be specified for all replication rules when there are multiple rules');
+    }
+    props.replicationRules.forEach(rule => {
+      if (rule.replicationTimeControl && !rule.metrics) {
+        throw new Error('\'replicationTimeControlMetrics\' must be enabled when \'replicationTimeControl\' is enabled.');
+      }
+      if (rule.deleteMarkerReplication && rule.tagFilter) {
+        throw new Error('\'tagFilter\' cannot be specified when \'deleteMarkerReplication\' is enabled.');
+      }
+    });
+
+    const destinationBuckets = props.replicationRules.map(rule => rule.destination.bucket);
+    const kmsKeys = props.replicationRules.map(rule => rule.kmsKey).filter(kmsKey => kmsKey !== undefined) as kms.IKey[];
+
+    const role = new iam.Role(this, 'ReplicationRole', {
+      assumedBy: new iam.ServicePrincipal('s3.amazonaws.com'),
+    });
+
+    // add permissions to the role
+    // @see https://docs.aws.amazon.com/AmazonS3/latest/userguide/setting-repl-config-perm-overview.html
+    role.addToPolicy(new iam.PolicyStatement({
+      actions: ['s3:GetReplicationConfiguration', 's3:ListBucket'],
+      resources: [Lazy.string({ produce: () => this.bucketArn })],
+      effect: iam.Effect.ALLOW,
+    }));
+    role.addToPolicy(new iam.PolicyStatement({
+      actions: ['s3:GetObjectVersionForReplication', 's3:GetObjectVersionAcl', 's3:GetObjectVersionTagging'],
+      resources: [Lazy.string({ produce: () => this.arnForObjects('*') })],
+      effect: iam.Effect.ALLOW,
+    }));
+    if (destinationBuckets.length > 0) {
+      role.addToPolicy(new iam.PolicyStatement({
+        actions: ['s3:ReplicateObject', 's3:ReplicateDelete', 's3:ReplicateTags', 's3:ObjectOwnerOverrideToBucketOwner'],
+        resources: destinationBuckets.map(bucket => bucket.arnForObjects('*')),
+        effect: iam.Effect.ALLOW,
+      }));
+    }
+
+    kmsKeys.forEach(kmsKey => {
+      kmsKey.grantEncrypt(role);
+    });
+
+    return {
+      role: role.roleArn,
+      rules: props.replicationRules.map((rule) => {
+        const sourceSelectionCriteria = (rule.replicaModifications !== undefined || rule.sseKmsEncryptedObjects !== undefined) ? {
+          replicaModifications: rule.replicaModifications !== undefined ? {
+            status: rule.replicaModifications ? 'Enabled' : 'Disabled',
+          } : undefined,
+          sseKmsEncryptedObjects: rule.sseKmsEncryptedObjects !== undefined ? {
+            status: rule.sseKmsEncryptedObjects ? 'Enabled' : 'Disabled',
+          } : undefined,
+        } : undefined;
+
+        // Whether to configure filter settings by And property.
+        const isAndFilter = rule.tagFilter && rule.tagFilter.length > 0;
+        // To avoid deploy error when there are multiple replication rules with undefined prefix,
```

Comment:
curious what deployment errors were observed in case of no prefix filter

Reply:
Not specifying a prefix filter means using the V1 replication configuration, which does not allow setting multiple replication rules.

To confirm this, I tried setting the filter property to undefined.

```ts
private renderWebsiteConfiguration(props: BucketProps): CfnBucket.WebsiteConfigurationProperty | undefined {
  ...
  return {
          id: rule.id,
          ...,
          // comment out filter option
          // filter,
        };
};
```

Result (`yarn integ test/aws-s3/test/integ.bucket-replication-test.js --update-on-failed`):
```sh
ReplicationStack | 4/7 | 12:58:45 PM | CREATE_FAILED        | AWS::S3::Bucket  | SourceBucket (SourceBucketDDD2130A) Resource handler returned message: "Number of distinct destination bucket ARNs cannot exceed 1 (Service: S3, Status Code: 400, Request ID: 56SVQEYYFXGSMG66, Extended Request ID: tAt6eHUYI0Hk5LuIaQQizGZDUE3sCLDI0dbcp8EMFU43ZKwqfF44lg6Q878Pz4hp8yjLzZDhnyQ=)" (RequestToken: db7089eb-d101-e4b7-51b4-7244fdc76577, HandlerErrorCode: InvalidRequest)
```

Reply:
This is `ReplicationRule.Prefix` property and it is different from `ReplicationRule.Filter.Prefix`.
By using the former, it becomes V1, and by using the latter, it becomes V2.